Shoko.Server/Utilities/MediaInfoLib/MenuBase64Converter.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Shoko.Server.Utilities.MediaInfoLib
+{
+    public class MenuBase64Converter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.IsAssignableFrom(typeof(Dictionary<string,string>));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.Null)
+                return null;
+            
+            var obj = JObject.Load(reader);
+
+            if (reader.Path != "extra") return obj.ToObject<Dictionary<string, string>>(); // Only continue if we are changing "extra" object in menu stream. Otherwise return as is.
+            
+            var sanitizedDictionary = new Dictionary<string, string>();
+
+            foreach (var (key, value) in obj)
+            {
+                switch (value?.Type)
+                {
+                    case JTokenType.String:
+                        sanitizedDictionary[key] = value?.ToString();
+                        break;
+                    case JTokenType.Object:
+                    {
+                        var base64EncodedBytes = Convert.FromBase64String(value?["#value"]?.ToString() ?? "");
+                        sanitizedDictionary[key] = Encoding.UTF8.GetString(base64EncodedBytes).Trim();
+                        break;
+                    }
+                    default:
+                        throw new NotImplementedException();
```

Comment:
Was not sure what to put for default case.

Reply:
Throw something more specific then a not implemented description, or at least provide a message

Reply:
If it's intentionally not implemented, I tend to go for a NotSupportedException

Reply:
I changed it to
`throw new NotSupportedException("Unknown type encountered in MenuStream.extra");`